sdk/servicebus/azure-servicebus/tests/perf_tests/T1_legacy_tests/_test_base.py

```diff
@@ -0,0 +1,139 @@
+
+from azure_devtools.perfstress_tests import PerfStressTest, get_random_bytes
+
+from azure.servicebus import ServiceBusClient, ReceiveSettleMode, Message
+from azure.servicebus.aio import ServiceBusClient as AsyncServiceBusClient
+from azure.servicebus.control_client import ServiceBusService
+from azure.servicebus.control_client.models import Queue
+
+
+def parse_connection_string(conn_str):
+    conn_settings = [s.split("=", 1) for s in conn_str.split(";")]
+    conn_settings = dict(conn_settings)
+    shared_access_key = conn_settings.get('SharedAccessKey')
+    shared_access_key_name = conn_settings.get('SharedAccessKeyName')
+    endpoint = conn_settings.get('Endpoint')
+    parsed = urlparse(endpoint.rstrip('/'))
+    namespace = parsed.netloc.strip().split('.')[0]
+    return {
+        'namespace': namespace,
+        'endpoint': endpoint,
+        'entity_path': conn_settings.get('EntityPath'),
+        'shared_access_key_name': shared_access_key_name,
+        'shared_access_key': shared_access_key
+    }
+
+
+class _ServiceTest(PerfStressTest):
+    service_client = None
+    async_service_client = None
+
+    def __init__(self, arguments):
+        super().__init__(arguments)
+
+        connection_string = self.get_from_env("AZURE_SERVICEBUS_CONNECTION_STRING")
+        if not _ServiceTest.service_client or self.args.service_client_per_instance:
+            _ServiceTest.service_client = ServiceBusClient.from_connection_string(connection_string)
+            _ServiceTest.async_service_client = AsyncServiceBusClient.from_connection_string(connection_string)
+
+        self.service_client = _ServiceTest.service_client
+        self.async_service_client =_ServiceTest.async_service_client
+
+    @staticmethod
+    def add_arguments(parser):
+        super(_ServiceTest, _ServiceTest).add_arguments(parser)
+        parser.add_argument('--message-size', nargs='?', type=int, help='Size of a single message. Defaults to 100 bytes', default=100)
+        parser.add_argument('--service-client-per-instance', action='store_true', help='Create one ServiceClient per test instance.  Default is to share a single ServiceClient.', default=False)
+        parser.add_argument('--num-messages', nargs='?', type=int, help='Number of messages to send or receive. Defaults to 100', default=100)
+
+
+class _QueueTest(_ServiceTest):
+    queue_name = "perfstress-" + str(uuid.uuid4())
+
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        connection_string = self.get_from_env("AZURE_SERVICEBUS_CONNECTION_STRING")
+        connection_props = parse_connection_string(connection_string)
+        self.mgmt_client = ServiceBusService(
+            service_namespace=connection_props['namespace'],
+            shared_access_key_name=connection_props['shared_access_key_name'],
+            shared_access_key_value=connection_props['shared_access_key']
+        )
+        queue = Queue(max_size_in_megabytes=40960)
```

Comment:
I think this code should move to `global_setup()`, in part because ctors should not contain code which makes network requests, and also for symmetry with `global_cleanup()`.